### ISSUE TITLE
False negative for unimplemented abstract method for classes decorated with @final

### ISSUE THREAD

**Issue by vepain** [astral-sh/ty#2525]
### Summary

ty does not return any error when a class decorated with `@final` does not implement a inherited abstract method.

The same false negative is observed when class `A` is a `Protocol`.

```py
from abc import ABC, abstractmethod
from typing import final

class A(ABC):
    @abstractmethod
    def foo(self) -> int:
        raise NotImplementedError

@final 
class B(A):  # should error: abstract method foo not implemented
    pass

if __name__ == "__main__":
    b = B()  # should also error 
    b.foo()  # also?
```

### Version

* Python 3.13
* ruff 0.14.13
* ty 0.0.12

**Comment by vepain** [comment]
This issue is potentially implied by this one: https://github.com/astral-sh/ty/issues/1877

**Comment by carljm** [comment]
Thanks! This is a known missing feature, but it doesn't look like we have an issue open for it yet, so we'll use this one.

(#1877 is a related but different feature.)
